client/state/themes/actions/theme-transfer.js

```diff
@@ -12,13 +12,34 @@ import {
 
 import 'calypso/state/themes/init';
 
+function initiateTransfer( siteId, plugin, theme, onProgress ) {
+	return new Promise( ( resolve, rejectPromise ) => {
+		const resolver = ( error, data ) => {
+			error ? rejectPromise( error ) : resolve( data );
+		};
+
+		const post = {
+			path: `/sites/${ siteId }/automated-transfers/initiate`,
+		};
+
+		if ( plugin ) {
+			post.body = { plugin };
+		}
+		if ( theme ) {
+			post.formData = [ [ 'theme', theme ] ];
+		}
+
+		const req = wpcom.req.post( post, resolver );
+		req && ( req.upload.onprogress = onProgress );
```

Comment:
`req` is guaranteed to be an XHR-like object, at other similar places we don't check it.

Reply:
Good point. Cleaned up in ea1762456f8a8175c16ef4d9e1ab84946fcb9c15

Reply:
Reverted that change. It was causing tests to fail, so I prefer keeping it.

Reply:
I'm pretty sure that it's a bug in the test, but let's solve that separately :+1:

Reply:
I agree that's very likely. I haven't investigated further, but it's definitely worth looking at.